### ISSUE TITLE
False positive for simplifiable-if-statement when the boolean values are not the same

### ISSUE THREAD

**Issue by PCManticore** [PyCQA/pylint#1984]
### Steps to reproduce
1. Save the following in a file and run pylint over it
```
A = True
B = 1
C = [1]
D = False
E = False

if A and B in C:
    D = True
else:
    E = True

print(D, E)
```

### Current behavior
pylint emits a `simplifiable-if-statement`.

### Expected behavior
pylint shouldn't emit `simplifiable-if-statement` since we assign to two different values in the `if` statement.

### pylint --version output
master & 1.8.3
